### PR TITLE
[EGD-6823] Add quote randomization

### PR DESCRIPTION
**Pull request by SP2FET**
Added randomization of the quotes when quote wallpaper is selected

### CODE REVIEW

**Review by wojtekrzepecki — APPROVED** [review]
looks good

File: module-services/service-db/agents/quotes/RandomizedQuoteModel.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                return;
+            }
+        }
+        settings->setValue(
+            settings::Quotes::randomQuotesList, serializer->serialize(list), settings::SettingsScope::Global);
+        LOG_ERROR("ID queue length: %zu", list.size());
+    }
+    auto RandomizedQuoteModel::getId() -> int
+    {
+
```
Comment (mjan-dev) [inline module-services/service-db/agents/quotes/RandomizedQuoteModel.cpp:49]: minor: extra newline

File: module-services/service-db/include/service-db/QuotesMessages.hpp [hunk 2]
```diff
@@ -387,6 +387,37 @@ namespace Quotes
             }
         };
 
+        class ReadRandomizedQuoteRequest : public db::Query
+        {
+          public:
+            explicit ReadRandomizedQuoteRequest() : Query(Query::Type::Read)
```
Comment (mjan-dev) [inline module-services/service-db/include/service-db/QuotesMessages.hpp:393]: explicit for the default constructor (no args) has no effect

File: products/PurePhone/services/appmgr/CMakeLists.txt [hunk 3]
```diff
@@ -5,7 +5,7 @@ target_sources(appmgr
         ApplicationManager.cpp
         models/WallpaperDisplayModel.cpp
         include/appmgr/models/WallpaperDisplayModel.hpp
-    PUBLIC
+        PUBLIC
```
Comment (mjan-dev) [inline products/PurePhone/services/appmgr/CMakeLists.txt:8]: some wrong auto formatting?

**Review by mjan-dev — APPROVED** [review]
Approved with minor comments.

File: module-apps/apps-common/popups/presenter/QuotesPresenter.cpp [hunk 4]
```diff
@@ -15,17 +15,16 @@ namespace gui
 
     void QuotesPresenter::requestQuote()
     {
-        // TODO: get quote: only enabled, from currently selected language
-        constexpr auto quoteID = 1;
-        auto query             = std::make_unique<Quotes::Messages::ReadQuoteRequest>(quoteID);
+        // TODO: get quote: from currently selected language
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply module-apps/apps-common/popups/presenter/QuotesPresenter.cpp:18]: removed

File: module-services/service-db/agents/quotes/QuotesAgent.cpp [hunk 5]
```diff
@@ -150,11 +160,25 @@ namespace Quotes
...
+        response->setRequestQuery(query);
+        return response;
+    }
+
     auto QuotesAgent::handleWriteQuote(std::shared_ptr<db::Query> query) -> std::unique_ptr<db::QueryResult>
     {
         auto req         = std::dynamic_pointer_cast<Messages::WriteQuoteRequest>(query);
         auto queryResult = database->execute(
             Queries::writeQuote, req->langId, req->quote.c_str(), req->author.c_str(), req->enabled, req->quoteId);
+        randomizedQuoteModel.updateList(true);
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply module-services/service-db/agents/quotes/QuotesAgent.cpp:181]: the parameter specifies if the update is forced or not, I'll add a variable to be more clear ;)

File: module-services/service-db/agents/quotes/RandomizedQuoteModel.cpp [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    void RandomizedQuoteModel::randomize(IdList &list)
+    {
+        srand(time(NULL));
+        std::random_shuffle(std::begin(list), std::end(list));
+    }
+    void RandomizedQuoteModel::updateList(bool forced)
+    {
+        auto queryResult = quotesDB->query(Queries::getEnabledQuotes);
+        auto quotesList  = getList<QuotesList, QuoteRecord>(0, 1000, std::move(queryResult));
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply module-services/service-db/agents/quotes/RandomizedQuoteModel.cpp:26]: fixed

File: module-services/service-db/agents/quotes/RandomizedQuoteModel.cpp (hunk 1 above)
Comment (SP2FET) [reply module-services/service-db/agents/quotes/RandomizedQuoteModel.cpp:49]: added

File: module-services/service-db/include/service-db/QuotesMessages.hpp (hunk 2 above)
Comment (SP2FET) [reply module-services/service-db/include/service-db/QuotesMessages.hpp:393]: true, leftover after copying

Comment (SP2FET) [reply module-services/service-db/include/service-db/QuotesMessages.hpp:393]: removed

File: products/PurePhone/services/appmgr/CMakeLists.txt (hunk 3 above)
Comment (SP2FET) [reply products/PurePhone/services/appmgr/CMakeLists.txt:8]: probably, fixed ;)

**Pull request merged**